README.md

```diff
@@ -486,6 +486,152 @@ rescue Stormpath::Error => e
+a protected resource (e.g. your application). Your application, with the help of Stormpath, then confirms the users 
+identity in order to determine whether they are able to access the resource. In SAML terminology, the user is the User 
+Agent, your application (along with Stormpath) is the Service Provider, and the third-party SAML authentication site 
+is the Identity Provider or IdP.
+
+The broad strokes of the process are as follows:
+
+* User Agent requests access from Service Provider
+* Service Provider responds with redirect to Identity Provider
+* Identity Provider authenticates the user
+* Identity provider redirects user back to Service Provider along with SAML assertions.
+* Service Provider receives SAML assertions and either creates or retrieves Account information
+
+#### Configuring Stormpath as a Service Provider
+
+Configuration is stored in the Directory's Provider resource. Both of these resources must also be linked with an 
+AccountStoreMapping. Here we will explain to you the steps that are required to configure Stormpath as a SAML Service 
+Provider.
+
+#### Step 1: Gather IDP Data
+
+You will need the following information from your IdP:
+
+- **SSO Login URL** - The URL at the IdP to which SAML authentication requests should be sent. This is often called an "SSO URL", "Login URL" or "Sign-in URL".
+- **SSO Logout URL** - The URL at the IdP to which SAML logout requests should be sent. This is often called a "Logout URL", "Global Logout URL" or "Single Logout URL".
+- **Signing Cert** - The IdP will digitally sign auth assertions and Stormpath will need to validate the signature. This will usually be in .pem or .crt format, but Stormpath requires the text value.
+- **Signing Algorithm** - You will need the name of the signing algorithm that your IdP uses. It will be either "RSA-SHA256" or "RSA-SHA1".
+
+#### Step 2: Configure Your SAML Directory
+
+Input the data you gathered in Step 1 above into your Directory's Provider resource, and then pass that along as part of the Directory creation HTTP POST:
+
+```ruby
+directory.provider
+provider = "saml"
+request = Stormpath::Provider::AccountRequest.new(provider, :access_token, access_token)
+application.get_provider_account(request)
+
+client.directories.create(
+  name: "infinum_directory", 
+  description: "random description", 
+  provider: {
+    provider_id: "saml"
+  }
+)
+```
+
+to get the data about the provider just type
+```ruby
+directory.provider
+```
+
+##### Retrive Your Service Provider Metadata
+
+Next you will have to configure your Stormpath-powered application as a Service Provider in your Identity Provider. This means that you will need to retrieve the correct metadata from Stormpath.
+
+In order to retrieve the required values, start by sending a GET to the Directory's Provider:
+
+```ruby
+directory.provider_metadata
+```
+
+From this metadata, you will need two values:
```

Comment:
Not sure if this section belongs here because we under the "Retrieve Your Service Provider Metadata" and sounds like we need something to actually retrieve the information.